client/rpc/src/eth.rs

```diff
@@ -511,10 +539,10 @@ impl<B, C, P, CT, BE, H: ExHashT> EthApiT for EthApi<B, C, P, CT, BE, H> where
 
 	fn transaction_count(&self, address: H160, number: Option<BlockNumber>) -> Result<U256> {
 		if let Some(BlockNumber::Pending) = number {
-			// Find future nonce
-			let id = BlockId::hash(self.client.info().best_hash);
-			let nonce: U256 = self.client.runtime_api()
-				.account_basic(&id, address)
+			let block = BlockId::Hash(self.client.info().best_hash);
+
```

Comment:
@tgmichel It's been a while since I've worked on this. I don't remember whether I removed this comment on purpose. Do you know anything about it?

Reply:
Probably not on purpose, the comment seems relevant to what happens https://github.com/paritytech/frontier/blob/8de4b12dd828bb1a462788f98dd9c5086b2325bb/client/rpc/src/eth.rs#L555